modules/asm/variables.tf

```diff
@@ -167,18 +167,28 @@ variable "ca" {
 
 variable "ca_certs" {
   description = "Sets CA certificate file paths when `ca` is set to `citadel`. These values must be provided when using Citadel as CA. Additional documentation on Citadel is available at https://cloud.google.com/service-mesh/docs/scripted-install/gke-install#installation_with_citadel_as_the_ca."
-  type        = map
+  type        = map(any)
   default     = {}
   # default = {
   #   "ca_cert"    = "none"
   #   "ca_key"     = "none"
   #   "root_cert"  = "none"
   #   "cert_chain" = "none"
   # }
+  validation {
+    condition     = contains([4, 0], length(compact([for k in ["ca_cert", "ca_key", "root_cert", "cert_chain"] : lookup(var.ca_certs, k, "")])))
+    error_message = "One or more maps for ca_certs are missing. If you plan to use the self-signed certificate, do not declare the ca_certs variable."
```

Comment:
```suggestion
    error_message = "One or more required keys for ca_certs are missing. If you plan to use the self-signed certificate, do not declare the ca_certs variable."
```

Reply:
I added this change in my one of my last commits.